RELEASE_NOTES.md

```diff
@@ -16,6 +16,7 @@ Further information, how to activate the debug mode of forked JVMs for the plug-
 - [Tycho-Surefire](https://www.eclipse.org/tycho/sitedocs/tycho-surefire-plugin/test-mojo.html#debugPort)
 - [Tycho-Eclipserun](https://www.eclipse.org/tycho/sitedocs/tycho-extras/tycho-eclipserun-plugin/eclipse-run-mojo.html#jvmArgs)
 
+This works for ordinary Maven projects as well as for Eclipse Plug-in projects build with Eclipse Tycho.
```

Comment:
This does not sound correct. There are no 'ordinary' projects compared to tycho projects because they are both "maven" builds.

I would suggest to either remove this or explicitly name the mojos of what build output we support.

From the other changes it seems more that you try to support pomless builds here and those without a maven nature. In this case it has nothing to do with building with tycho I think but more with finding sources in non-m2e projects.
A better aproach might be to use **JDT** to dearch fir the type instead of searching for a certain project making assumptions about the artifact-id.

Reply:
> This does not sound correct. There are no 'ordinary' projects compared to tycho projects because they are both "maven" builds.
> 
> I would suggest to either remove this or explicitly name the mojos of what build output we support.

This list was actually only intended as a list of examples respectively a reference for the (IMHO) most common use cases. All kind of plug-ins that launch other VMs are supported as long as one knows the right properties to command the plug-in to launch the VM the with corresponding debugger flags. The text `Listening for transport dt_socket at address:` is emitted by the JVM (you can see it also wenn launching JVM with corresponding flags manually) so I think this is quite generic.

But I think I will just remove the remark. As long as it works people are usually happy and I assume most wont except a difference in the first place any ways.

> 
> From the other changes it seems more that you try to support pomless builds here and those without a maven nature. In this case it has nothing to do with building with tycho I think but more with finding sources in non-m2e projects. A better aproach might be to use **JDT** to dearch fir the type instead of searching for a certain project making assumptions about the artifact-id.

The root problem is to find out which project from the workspace the print-out refers to. And since Maven runs in a separate process the information is not available in the Eclipse VM.
The cleanest way would probably to add an Event-Spy via a Maven-Extension to the Maven-VM that connects to the Eclipse VM and reports the information need here to the Eclipse-VM, namely groupId, artifactId and version as well as the project location. The latter can be used to search an `IProject` by its URI using the ResourcePlug-ins workspace. The ConsoleLineTracker then just has to place the links at the right text.

Igor Fedorenko has done something similar for the MavenDev view:
https://github.com/ifedorenko/com.ifedorenko.m2e.mavendev/blob/master/com.ifedorenko.m2e.mavendev.launch.ui.eventspy/src/main/java/com/ifedorenko/m2e/mavendev/launch/ui/eventspy/internal/ExecutionEventListener.java

But this would take a bit more time and I won't be able to complete this within the few next days.